pktoken/simplejws/simplejws.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 OpenPubkey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package simplejws
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/openpubkey/openpubkey/util"
+)
+
+type Jws struct {
+	Payload    string      `json:"payload"`    // Base64 encoded
+	Signatures []Signature `json:"signatures"` // Base64 encoded
+}
+type Signature struct {
+	Protected string                 `json:"protected"` // Base64 encoded
+	Public    map[string]interface{} `json:"header,omitempty"`
+	Signature string                 `json:"signature"` // Base64 encoded
+}
+
+func (s *Signature) GetTyp() (string, error) {
+	decodedProtected, err := util.Base64DecodeForJWT([]byte(s.Protected))
+	if err != nil {
+		return "", err
+	}
+	type protectedTyp struct {
+		Typ string `json:"typ"`
+	}
+	var ph protectedTyp
+	err = json.Unmarshal(decodedProtected, &ph)
+	if err != nil {
+		return "", err
+	}
+	return ph.Typ, nil
+}
+
+type SigOptStruct struct {
+	PublicHeader map[string]any
+}
+type SigOpts func(a *SigOptStruct)
+
+// WithPublicHeader species that a public header be included in the
+// signature. Public headers aren't Base64 encoded because they aren't signed.
+// Example use: WithPublicHeader(map[string]any{"key1": "abc", "key2": "def"})
+func WithPublicHeader(publicHeader map[string]any) SigOpts {
+	return func(o *SigOptStruct) {
+		o.PublicHeader = publicHeader
+	}
+}
+
+func (j *Jws) AddSignature(token []byte, opts ...SigOpts) error {
+	sigOpts := &SigOptStruct{}
+	for _, applyOpt := range opts {
+		applyOpt(sigOpts)
+	}
+
+	protected, payload, signature, err := jws.SplitCompact(token)
```

Comment:
```suggestion
	protected, payload, signature, err := SplitCompact(token)
```
to remove `github.com/lestrrat-go/jwx/v2/jws` dependency